test/integration/rmi_out_test.py

```diff
@@ -43,27 +51,146 @@ def test_deprish_out(rmi_out, request):
+    the FERC-EIA-Depreciation tables we need to add *both* versions of the
+    depreciation `data_source`.
+    """
+    if 'data_source' not in df:
+        df = pd.concat([
+            df.assign(data_source='PUC'),
+            df.assign(data_source='FERC'),
+        ])
+    return df
+
+
+@pytest.mark.parametrize(
+    "df1_name,df2_name,data_col,by_name", [
+        (
+            "grab_deprish",
+            "grab_deprish_to_eia",
+            'plant_balance_w_common',
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_eia",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities',
+        ),
+        (
+            "grab_deprish_to_eia",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'utilities'
+        ),
+        (
+            "grab_deprish",
+            "grab_deprish_to_ferc1",
+            'plant_balance_w_common',
+            'plants',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'utilities',
+        ),
+        (
+            "grab_ferc1_to_eia",
+            "grab_deprish_to_ferc1",
+            'capex_total',
+            'plants',
+        ),
+    ]
+)
+def test_data_col_at_two_stages(
+    rmi_out,
+    df1_name: str,
+    df2_name: str,
+    data_col: str,
+    by_name: Literal['plants', 'utilities'],
+):
+    """
+    Test the consistency of a data column at two stages.
+
+    The data that is processed in this repo goes along multiple stages of its
+    journey. There are some quantities that are expected to be invariant when
+    aggregated at the plant level, and the utility level -- that the quantity
+    might slosh around between generators in a plant, or between plants within
+    a utility, but that when aggregated to the plant or utility level, it
+    should be the same at every step in the processing.
+
+    Ideally, the inputs
+    """
+    if by_name == 'plants':
+        by = PK_PLANTS
+    elif by_name == 'utilities':
+        by = PK_UTILS
+
+    test_stages = pudl_rmi.connect_deprish_to_ferc1.gb_test(
+        df1=rmi_out.__getattribute__(df1_name)().pipe(_add_data_source),
+        df2=rmi_out.__getattribute__(df2_name)().pipe(_add_data_source),
+        data_col=data_col,
+        by=by
+
+    )
+    # we add the off_rate.notnll() mask in here bc there are a ton of unconnected
+    # data in these tables (eg. depreciation records that don't have a EIA connection)
+    # these bbs will not get effectively propegated through each of these stages that
```

Comment:
What does "bbs" mean here? What is `off_rate`? I'm not understanding why that column in particular is important. Does it have to do with electricity rates or depreciation rates, and it's only present in rows that have data?

Could you also do something like:
```py
fails = test_stages.dropna(subset=["off_rate"])[~test_stages.match]
```

Especially without `gb_test` being in this module, it's not clear what it's returning -- what kind of thing is `test_stages`? It all feels kind of cryptic. A short explanation of what's happening / expected would be great.

Also what does it "fails" mean? There are a lot of different kinds of failures.

Reply:
bbs = records here... i'll make it clear and less fun =(

but really the `off_rate` being null is an indication that either df1 or df2 has a null `data_col`. I'll move `gb_test` in here and make this note more clear. I also included more explainations of the nulls in the test doc string

Reply:
I'd personally rather keep the mask version as oppose to the dropna... It looks cleaner to me.

But does it feel okay with the updated docs and comments?

Reply:
and fails here is... plants or utilities that have non-null but inconsistent values for the `data_col`... if you have a clearer but concise name for that plz halp!

Reply:
How about  `actual_unequal_rows` and then below use `expected_unequal_rows`

Reply:
that makes a lot of sense and feels very clear to me